app/controllers/api/v0/plans_controller.rb

```diff
@@ -61,10 +61,44 @@ def index
     if params[:per_page].present? && params[:per_page].to_i > Rails.configuration.branding[:application][:api_max_page_size]
       params[:per_page] = Rails.configuration.branding[:application][:api_max_page_size]
     end
-    @plans = paginate @user.org.plans.includes( [ {answers: :question_options} ,
-      template: [ { phases: { sections: { questions: :question_format } } }, :org]
+    @plans = paginate @user.org.plans.includes( [ {roles: :user}, {answers: :question_options} ,
+      template: [ { phases: { sections: { questions: [:question_format, :themes ] } } }, :org]
       ])
+    
+    # Filter on list of users
+    user_ids = extract_param_list(params, "user")
+    @plans = @plans.where(roles: {user_id: user_ids}) if user_ids.present?
+    # filter on dates
+    if params["created_after"].present? || params["created_before"].present?
+      @plans = @plans.where(created_at: dates_to_range(params,"created_after","created_before"))
+    end
+    if params["updated_after"].present? || params["updated_before"].present?
+      @plans = @plans.where(updated_at: dates_to_range(params,"updated_after","updated_before"))
+    end
+    # filter on funder (dmptemplate_id)
+    template_ids = extract_param_list(params, "template")
```

Comment:
might be more efficient to process template and plan id filtering before the dates

Reply:
I think it should be lazy-evaluating, so activerecord should apply all stages of filtering at the same time.